src/test/java/software/amazon/awssdk/crt/test/SigningTest.java

```diff
@@ -271,6 +274,53 @@ public Credentials getCredentials() {
         }
         assertTrue(failed);
     }
+    @Test
+    public void testSigningAnonymous() throws Exception {
+            HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+            try (AwsSigningConfig config = new AwsSigningConfig()) {
+                config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+                config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+                config.setRegion("us-east-1");
+                config.setService("service");
+                config.setTime(System.currentTimeMillis());
+                config.setUseDoubleUriEncode(true);
+                config.setShouldNormalizeUriPath(true);
+                config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+                config.setCredentials(Credentials.createAnonymousCredentials());
+
+                CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+                HttpRequest signedRequest = result.get();
+                assertNotNull(signedRequest);
+                assertFalse(hasHeader(signedRequest, "Authorization"));
```

Comment:
trivial: assert that the original request and the signed request have the same number of headers